---
title: "CRISPR-guided multiomics unveils direct mTOR inhibition of Hippo signaling through CDCP1: Summary of Transcriptomics / Phosphoproteomics analyses"
output:
  html_document:
    df_print: kable
    toc: true
    toc_depth: 3
  pdf_document:
    df_print: kable
    toc: true
    toc_depth: 3
    latex_engine: xelatex
---

# Overview

Here, differential abundance results summarized for each of the mTOR complex
knockouts, for both the transcriptome and phosphoproteome datasets.

DESeq2 was used to assess differential expression at the transcriptome level, while Proteome
Discoverer was used to detect differerences across conditions at the phosphoproteome level.

In order to attempt to understand how things are changing across the different levels of expression
and regulation, proteins described in the phosphoproteome data are mapped to putative genes using an
ID mapping from Uniprot.

_Note_: in the summary tables below, values less than 1e-11 will be printined as "0"'s. For the
precise values, refer to the actual results tables provided in a separate supplemental file.

# Methods & Results

## Setup

```{r load_libraries, message = FALSE}
library(annotables)
library(tidyverse)
library(yaml)
library(RColorBrewer)
library(openxlsx)
library(knitr)
library(VennDiagram)

# be more precise when printing summary tables
options(digits = 10)

# plot opts
knitr::opts_chunk$set(
  fig.width = 6,
  fig.height = 3.6,
  fig.retina = 1,
  dpi = 192
)

if (opts_knit$get("rmarkdown.pandoc.to") == "latex") {
  opts_chunk$set(dev = "cairo_pdf", dev.args = list(cairo_pdf = list(family = "Times New Roman")))
}

# volcano plot x & y axes limits
TX_XLIMS <- list(-10, 10)
TX_YLIMS <- list(0, 160)

PHOSPHO_XLIMS <- list(-3, 3)
PHOSPHO_YLIMS <- list(0, 14)
```

```{r helper_func}
#
# plot_volcano
#
# Helper function to generate a static or interactive volcano plot
#
plot_volcano <- function(dat, mtorc1_genes, mtorc2_genes, mtorc3_genes, target, plot_title,
                         xlims, ylims, data_type = "tx") {
  dat <- dat %>%
    mutate(log10pval = -log10(padj))

  # rename for consistency
  if (data_type == "tx") {
    dat <- dat %>%
      rename(log2ratio = log2FoldChange)
  }

  # drop genes with missing values
  dat <- dat[!is.na(dat$log2ratio) & !is.na(dat$log10pval), ]

  # assign specific colors to highlight genes significantly modulated in each of the mTOR complex
  # knockouts
  mtorc1_unique <- mtorc1_genes[!mtorc1_genes %in% c(mtorc2_genes, mtorc3_genes)]
  mtorc2_unique <- mtorc2_genes[!mtorc2_genes %in% c(mtorc1_genes, mtorc3_genes)]
  mtorc3_unique <- mtorc3_genes[!mtorc3_genes %in% c(mtorc1_genes, mtorc2_genes)]

  shared_12 <- intersect(mtorc1_genes, mtorc2_genes)
  shared_13 <- intersect(mtorc1_genes, mtorc3_genes)
  shared_23 <- intersect(mtorc2_genes, mtorc3_genes)
  shared_123 <- intersect(mtorc1_genes, shared_23)

  # complex name? (eg. "mTORC1")
  complex_name <- ifelse(target == "rap", "mTORC1", ifelse(target == "ric", "mTORC2", "mTORC3"))

  # assign colors
  pal <- brewer.pal(4, "Dark2")
  names(pal) <- c("mTORC1/2", "mTORC1/3", "mTORC2/3", "mTORC1-3")

  pal <- c(pal, "#7F7F7F", "#AAAAAA")
  names(pal)[(length(pal) - 1):(length(pal))] <- c(complex_name, "Not significant")

  dat$group <- "Not significant"
  dat$group[dat$padj < 0.01] <- complex_name

  if (target == "rap") {
    dat$group[dat$symbol %in% shared_12] <- "mTORC1/2"
    dat$group[dat$symbol %in% shared_13] <- "mTORC1/3"
    pal <- pal[-3]
  } else if (target == "ric") {
    dat$group[dat$symbol %in% shared_12] <- "mTORC1/2"
    dat$group[dat$symbol %in% shared_23] <- "mTORC2/3"
    pal <- pal[-2]
  } else if (target == "meak7") {
    dat$group[dat$symbol %in% shared_13] <- "mTORC1/3"
    dat$group[dat$symbol %in% shared_23] <- "mTORC2/3"
    pal <- pal[-1]
  }

  dat$group[dat$symbol %in% shared_123] <- "mTORC1-3"

  # interactive plot (plotly)
  if (opts_knit$get("rmarkdown.pandoc.to") == "html") {
    library(plotly)

    # helper func to draw vertical lines
    # https://stackoverflow.com/a/34097929/554531
    vline <- function(x = 0, color = "red") {
      list(type = "line", y0 = 0, y1 = 1, yref = "paper", x0 = x, x1 = x,
           line = list(color = color, dash = "dashed", width = 1))
    }

    if (data_type == "tx") {
      hover_text <- ~paste("<b>", symbol, "</b><br><br>",
                           ensgene, "<br><br>",
                           "Signifiance: ", group, "<br><br>",
                           "<b>Adj. P-val:</b> ", padj, "<extra></extra>")
    } else {
      hover_text <- ~paste("<b>", symbol, "</b><br>",
                           uniprot, "<br><br>",
                           "Significance: ", group, "<br><br>",
                           "<b>Adj. P-val:</b> ", padj, "<extra></extra>")
    }

    plot_ly(data = dat,
            x = ~log2ratio,
            y = ~log10pval,
            color = ~group,
            colors = pal,
            type = "scatter",
            mode = "markers",
            marker = list(size = 5),
            text = hover_text) %>%
      layout(xaxis = list(range = xlims), yaxis = list(range = ylims)) %>%
      layout(title = plot_title) %>%
      layout(shapes = list(vline(-1), vline(1))) %>%
      add_trace(type = "scatter", hovertemplate = "%{text}", showlegend = FALSE)

  } else {
    # static plot (ggplot2)
    ggplot(data = dat, aes(x = log2ratio, y = log10pval, color = group)) +
      geom_point() +
      scale_colour_manual(name = "Significance", values = pal) +
      ggtitle(plot_title) +
      geom_vline(xintercept = c(-1, 1), color = "#FF4444") +
      xlim(unlist(xlims)) +
      ylim(unlist(ylims)) +
      theme_bw()
  }
}
```

```{r load_config}
cfg <- yaml::read_yaml("config/config.yml")

if (!dir.exists(cfg$output_dir)) {
  dir.create(cfg$output_dir, recursive = TRUE, mode = "0755")
}
```

```{r load_uniprot_id_mapping}
# load uniprot id mapping
mapping <- read_tsv(cfg$uniprot_mapping,
                    col_names = c("uniprotkb_ac", "field", "value"),
                    show_col_types = FALSE)

# limit to gene symbols and convert from long -> wide
mapping <- mapping %>%
  filter(field == "Gene_Name") %>%
  select(uniprotkb_ac, symbol = value)

# drop the relatively small number of multimapped entries
mapping <- mapping[!duplicated(mapping$uniprotkb_ac), ]
mapping <- mapping[!grepl(";", mapping$symbol), ]
```

```{r load_tx_data}
deseq_dir <- file.path(cfg$data_dirs$rna, "deseq2")

# load transcriptome deseq2 results for each of knockout and each nutrient condition
deseq_nutri <- list(
  "rap" = read_tsv(file.path(deseq_dir, "condition_Rap_nutri_vs_v_nutri.tsv"), show_col_types = FALSE),
  "ric" = read_tsv(file.path(deseq_dir, "condition_Ric_nutri_vs_v_nutri.tsv"), show_col_types = FALSE),
  "meak7" = read_tsv(file.path(deseq_dir, "condition_m7_nutri_vs_v_nutri.tsv"), show_col_types = FALSE)
)

deseq_starved <- list(
  "rap" = read_tsv(file.path(deseq_dir, "condition_Rap_starved_vs_v_starved.tsv"), show_col_types = FALSE),
  "ric" = read_tsv(file.path(deseq_dir, "condition_Ric_starved_vs_v_starved.tsv"), show_col_types = FALSE),
  "meak7" = read_tsv(file.path(deseq_dir, "condition_m7_starved_vs_v_starved.tsv"), show_col_types = FALSE)
)

# compute average p-values across each pair of contrasts as well as the average for all three
rap <- deseq_nutri[["rap"]] %>%
  select(ensgene, symbol, padj_rap = padj)

ric <- deseq_nutri[["ric"]] %>%
  select(ensgene, padj_ric = padj)

meak7 <- deseq_nutri[["meak7"]] %>%
  select(ensgene, padj_meak7 = padj)

deseq_nutri[["combined"]] <- rap %>%
  inner_join(ric, by = "ensgene") %>%
  inner_join(meak7, by = "ensgene") %>%
  na.omit() %>%
  mutate(
    padj_rap_ric = (padj_rap + padj_ric) / 2,
    padj_rap_meak7 = (padj_rap + padj_meak7) / 2,
    padj_ric_meak7 = (padj_ric + padj_meak7) / 2,
    padj_all = (padj_rap + padj_ric + padj_meak7) / 3
  ) %>%
  arrange(padj_all)

# same thing, but for the starved data
rap <- deseq_starved[["rap"]] %>%
  select(ensgene, symbol, padj_rap = padj)

ric <- deseq_starved[["ric"]] %>%
  select(ensgene, padj_ric = padj)

meak7 <- deseq_starved[["meak7"]] %>%
  select(ensgene, padj_meak7 = padj)

deseq_starved[["combined"]] <- rap %>%
  inner_join(ric, by = "ensgene") %>%
  inner_join(meak7, by = "ensgene") %>%
  na.omit() %>%
  mutate(
    padj_rap_ric = (padj_rap + padj_ric) / 2,
    padj_rap_meak7 = (padj_rap + padj_meak7) / 2,
    padj_ric_meak7 = (padj_ric + padj_meak7) / 2,
    padj_all = (padj_rap + padj_ric + padj_meak7) / 3
  ) %>%
  arrange(padj_all)
```

```{r load_phospho_nutri_data}
# load phospho nutrient-fed proteome discoverer results
phospho_nutri_infile <- file.path(cfg$data_dirs$phospho, "phospho-peptides-nutri.tsv")

phospho_nutri <- read_tsv(phospho_nutri_infile, show_col_types = FALSE) %>%
  select(uniprot = `Master Protein Accessions`,
         mod = Modifications,
         mod_proteins = `Modifications in Master Proteins`,
         rap_log2ratio = `Abundance Ratio (log2): (Rap_del_1R) / (V_del_4R)`,
         rap_padj = `Abundance Ratio Adj. P-Value: (Rap_del_1R) / (V_del_4R)`,
         rap_pval = `Abundance Ratio P-Value: (Rap_del_1R) / (V_del_4R)`,
         ric_log2ratio = `Abundance Ratio (log2): (Ric_del_SR) / (V_del_4R)`,
         ric_padj = `Abundance Ratio Adj. P-Value: (Ric_del_SR) / (V_del_4R)`,
         ric_pval = `Abundance Ratio P-Value: (Ric_del_SR) / (V_del_4R)`,
         meak7_log2ratio = `Abundance Ratio (log2): (M7_del_SR) / (V_del_4R)`,
         meak7_padj = `Abundance Ratio Adj. P-Value: (M7_del_SR) / (V_del_4R)`,
         meak7_pval = `Abundance Ratio P-Value: (M7_del_SR) / (V_del_4R)`)

# add gene symbols
phospho_nutri$symbol <- mapping$symbol[match(phospho_nutri$uniprot, mapping$uniprotkb_ac)]

phospho_nutri <- phospho_nutri %>%
  select(uniprot, symbol, everything())

# limit to phospho entries
phospho_mask <- grepl("Phospho", phospho_nutri$mod)
table(phospho_mask)

phospho_nutri <- phospho_nutri[phospho_mask, ]

# add pairwise and three-way average p-values
phospho_nutri <- phospho_nutri %>%
  mutate(
    padj_rap_ric = (rap_padj + ric_padj) / 2,
    padj_rap_meak7 = (rap_padj + meak7_padj) / 2,
    padj_ric_meak7 = (ric_padj + meak7_padj) / 2,
    padj_all = (rap_padj + ric_padj + meak7_padj) / 3
  )

dim(phospho_nutri)
```

```{r load_phospho_starved_data}
# load phospho starved proteome discoverer results
phospho_starved_infile <- file.path(cfg$data_dirs$phospho, "phospho-peptides-starved.tsv")

phospho_starved <- read_tsv(phospho_starved_infile, show_col_types = FALSE) %>%
  select(uniprot = `Master Protein Accessions`,
         mod = Modifications,
         mod_proteins = `Modifications in Master Proteins`,
         rap_ratio = `Abundance Ratio: (Rap_del_1R) / (V_del_4R)`,
         rap_padj = `Abundance Ratio Adj. P-Value: (Rap_del_1R) / (V_del_4R)`,
         ric_ratio = `Abundance Ratio: (Ric_del_SR) / (V_del_4R)`,
         ric_padj = `Abundance Ratio Adj. P-Value: (Ric_del_SR) / (V_del_4R)`,
         meak7_ratio = `Abundance Ratio: (M7_del_SR) / (V_del_4R)`,
         meak7_padj = `Abundance Ratio Adj. P-Value: (M7_del_SR) / (V_del_4R)`)

# log2 transform abundance ratios to be consistent with the nutrient-fed sample results
phospho_starved <- phospho_starved %>%
  mutate(rap_log2ratio = log2(rap_ratio),
         ric_log2ratio = log2(ric_ratio),
         meak7_log2ratio = log2(meak7_ratio)) %>%
  select(-rap_ratio, -ric_ratio, -meak7_ratio)

# add gene symbols
phospho_starved$symbol <- mapping$symbol[match(phospho_starved$uniprot, mapping$uniprotkb_ac)]

phospho_starved <- phospho_starved %>%
  select(uniprot, symbol, everything())

# limit to phospho entries
phospho_mask <- grepl("Phospho", phospho_starved$mod)
table(phospho_mask)

phospho_starved <- phospho_starved[phospho_mask, ]

# add pairwise and three-way average p-values
phospho_starved <- phospho_starved %>%
  mutate(
    padj_rap_ric = (rap_padj + ric_padj) / 2,
    padj_rap_meak7 = (rap_padj + meak7_padj) / 2,
    padj_ric_meak7 = (ric_padj + meak7_padj) / 2,
    padj_all = (rap_padj + ric_padj + meak7_padj) / 3
  )

dim(phospho_starved)
```

```{r created_simplified_phospho_results}
# create simplified versions of Proteome Discoverer results
phospho_nutri_rap_full <- phospho_nutri %>%
  select(uniprot, symbol, mod_proteins, log2ratio = rap_log2ratio, pval = rap_pval, padj = rap_padj) %>%
  filter(!is.na(padj))

phospho_starved_rap_full <- phospho_starved %>%
  select(uniprot, symbol, mod_proteins, log2ratio = rap_log2ratio, padj = rap_padj) %>%
  filter(!is.na(padj))

phospho_nutri_ric_full <- phospho_nutri %>%
  select(uniprot, symbol, mod_proteins, log2ratio = ric_log2ratio, pval = ric_pval, padj = ric_padj) %>%
  filter(!is.na(padj))

phospho_starved_ric_full <- phospho_starved %>%
  select(uniprot, symbol, mod_proteins, log2ratio = ric_log2ratio, padj = ric_padj) %>%
  filter(!is.na(padj))

phospho_nutri_meak7_full <- phospho_nutri %>%
  select(uniprot, symbol, mod_proteins, log2ratio = meak7_log2ratio, pval = meak7_pval, padj = meak7_padj) %>%
  filter(!is.na(padj))

phospho_starved_meak7_full <- phospho_starved %>%
  select(uniprot, symbol, mod_proteins, log2ratio = meak7_log2ratio, padj = meak7_padj) %>%
  filter(!is.na(padj))
```

```{r collapse_phospho_sites}
# For each result set, create a table of the most significant phosphorylation sites associated with
# each gene
phospho_nutri_rap <- phospho_nutri_rap_full %>%
  group_by(symbol) %>%
  arrange(pval) %>%
  slice(1) %>%
  ungroup()

phospho_starved_rap <- phospho_starved_rap_full %>%
  group_by(symbol) %>%
  arrange(padj) %>%
  slice(1) %>%
  ungroup()

phospho_nutri_ric <- phospho_nutri_ric_full %>%
  group_by(symbol) %>%
  arrange(pval) %>%
  slice(1) %>%
  ungroup()

phospho_starved_ric <- phospho_starved_ric_full %>%
  group_by(symbol) %>%
  arrange(padj) %>%
  slice(1) %>%
  ungroup()

phospho_nutri_meak7 <- phospho_nutri_meak7_full %>%
  group_by(symbol) %>%
  arrange(pval) %>%
  slice(1) %>%
  ungroup()

phospho_starved_meak7 <- phospho_starved_meak7_full %>%
  group_by(symbol) %>%
  arrange(padj) %>%
  slice(1) %>%
  ungroup()
```

## ΔRPTOR nutrient-fed

### Transcriptome (ΔRPTOR nutrient-fed)

Genes with the highest differential expression between ΔRPTOR and control?

```{r tx_nutri_rap}
deseq_nutri[["rap"]] %>%
  arrange(padj) %>%
  head(cfg$table_max_rows)
```

```{r raptor_nutri_tx_volcano_plot, message = FALSE, warning = FALSE}
# significant genes for each complex knockout
mtorc1_tx_nutri_genes <- deseq_nutri[["rap"]] %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

mtorc2_tx_nutri_genes <- deseq_nutri[["ric"]] %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

mtorc3_tx_nutri_genes <- deseq_nutri[["meak7"]] %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

plot_volcano(deseq_nutri[["rap"]],
             mtorc1_tx_nutri_genes, mtorc2_tx_nutri_genes, mtorc3_tx_nutri_genes,
             "rap",
             "ΔRPTOR Transcriptome Differential Expression (nutrient-fed)",
             TX_XLIMS, TX_YLIMS)
```

### Phosphoproteome (ΔRPTOR nutrient-fed)

Phospho sites with the highest differential abundance between ΔRPTOR and control?

```{r phospho_nutri_rap_top_hits}
phospho_nutri_rap_full %>%
  arrange(padj) %>%
  head(cfg$table_max_rows)
```

How many proteins are associated with multiple phosphorylation sites?

```{r phospho_nutri_rap_multiple_phospho_sites}
table(duplicated(phospho_nutri_rap_full$symbol))
```

```{r raptor_nutri_volcano_plot, message = FALSE, warning = FALSE}
# significant genes for each complex knockout
mtorc1_phospho_nutri_genes <- phospho_nutri_rap %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

mtorc2_phospho_nutri_genes <- phospho_nutri_ric %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

mtorc3_phospho_nutri_genes <- phospho_nutri_meak7 %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

plot_volcano(phospho_nutri_rap,
             mtorc1_phospho_nutri_genes, mtorc2_phospho_nutri_genes, mtorc3_phospho_nutri_genes,
             "rap",
             "ΔRPTOR Phosphoproteome Differential Abundance (nutrient-fed)",
             PHOSPHO_XLIMS, PHOSPHO_YLIMS,
             "phospho")
```

### Phosphoproteome vs. transcriptome (ΔRPTOR nutrient-fed)

Are there genes which were found to be differentially expressed in the ΔRPTOR transcriptome data,
which also appear to encode proteins that are differentially phosphorylated?

```{r tx_vs_phospho}
phospho_nutri_rap$tx_pval <- deseq_nutri[["rap"]]$pvalue[match(phospho_nutri_rap$symbol, deseq_nutri[["rap"]]$symbol)]
phospho_nutri_rap$tx_padj <- deseq_nutri[["rap"]]$padj[match(phospho_nutri_rap$symbol, deseq_nutri[["rap"]]$symbol)]

# since there are a large number of significant genes in the tx results, rather than simply looking
# at the intersection of "significant" genes, gene p-value rankings between the tx/phospho data will
# be compared
phospho_nutri_rap$phospho_pval_rank <- rank(phospho_nutri_rap$pval)
phospho_nutri_rap$tx_pval_rank <- rank(phospho_nutri_rap$tx_pval)

phospho_nutri_rap <- phospho_nutri_rap %>%
  mutate(average_pval_rank = (phospho_pval_rank + tx_pval_rank) / 2)
```

Which genes are present and appear significant in both datasets?

```{r rap_nutri_common_sig}
phospho_nutri_rap %>%
  filter(!is.na(tx_pval) & !is.na(symbol)) %>%
  filter(padj <= 0.01 & tx_padj <= 0.01) %>%
  arrange(average_pval_rank) %>%
  head(cfg$table_max_rows)
```

## ΔRPTOR starved

### Transcriptome (ΔRPTOR starved)

Genes with the highest differential expression between ΔRPTOR and control?

```{r tx_starved_rap}
deseq_starved[["rap"]] %>%
  arrange(padj) %>%
  head(cfg$table_max_rows)
```

```{r raptor_starved_volcano_plot, message = FALSE, warning = FALSE}
# significant genes for each complex knockout
mtorc1_tx_starved_genes <- deseq_starved[["rap"]] %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

mtorc2_tx_starved_genes <- deseq_starved[["ric"]] %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

mtorc3_tx_starved_genes <- deseq_starved[["meak7"]] %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

plot_volcano(deseq_starved[["rap"]],
             mtorc1_tx_starved_genes, mtorc2_tx_starved_genes, mtorc3_tx_starved_genes,
             "rap",
             "ΔRPTOR Transcriptome Differential Expression (starved)",
             TX_XLIMS, TX_YLIMS)
```

### Phosphoproteome (ΔRPTOR starved)

Phospho sites with the highest differential abundance between ΔRPTOR and control?

```{r phospho_starved_rap_top_hits}
phospho_starved_rap_full %>%
  arrange(padj) %>%
  head(cfg$table_max_rows)
```

How many proteins are associated with multiple phosphorylation sites?

```{r phospho_starved_rap_multiple_phospho_sites}
table(duplicated(phospho_starved_rap_full$symbol))
```

```{r raptor_starved_volcano_plot, message = FALSE, warning = FALSE}
# significant genes for each complex knockout
mtorc1_phospho_nutri_genes <- phospho_nutri_rap %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

mtorc2_phospho_nutri_genes <- phospho_nutri_ric %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

mtorc3_phospho_nutri_genes <- phospho_nutri_meak7 %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

plot_volcano(phospho_starved_rap,
             mtorc1_phospho_nutri_genes, mtorc2_phospho_nutri_genes, mtorc3_phospho_nutri_genes,
             "rap",
             "ΔRPTOR Phosphoproteome Differential Abundance (starved)",
             PHOSPHO_XLIMS, PHOSPHO_YLIMS,
             "phospho")
```

### Phosphoproteome vs. transcriptome (ΔRPTOR starved)

Are there genes which were found to be differentially expressed in the ΔRPTOR transcriptome data,
which also appear to encode proteins that are differentially phosphorylated?

```{r tx_vs_phospho}
phospho_starved_rap$tx_pval <- deseq_starved[["rap"]]$pvalue[match(phospho_starved_rap$symbol, deseq_starved[["rap"]]$symbol)]
phospho_starved_rap$tx_padj <- deseq_starved[["rap"]]$padj[match(phospho_starved_rap$symbol, deseq_starved[["rap"]]$symbol)]

# since there are a large number of significant genes in the tx results, rather than simply looking
# at the intersection of "significant" genes, gene p-value rankings between the tx/phospho data will
# be compared
phospho_starved_rap$phospho_pval_rank <- rank(phospho_starved_rap$padj)
phospho_starved_rap$tx_pval_rank <- rank(phospho_starved_rap$tx_pval)

phospho_starved_rap <- phospho_starved_rap %>%
  mutate(average_pval_rank = (phospho_pval_rank + tx_pval_rank) / 2)
```

Which genes are present and appear significant in both datasets?

```{r rap_starved_common_sig}
phospho_starved_rap %>%
  filter(!is.na(tx_pval) & !is.na(symbol)) %>%
  filter(padj <= 0.01 & tx_padj <= 0.01) %>%
  arrange(average_pval_rank) %>%
  head(cfg$table_max_rows)
```

## ΔRICTOR nutrient-fed

### Transcriptome (ΔRICTOR nutrient-fed)

Genes with the highest differential expression between ΔRICTOR and control?

```{r tx_nutri_ric}
deseq_nutri[["ric"]] %>%
  arrange(padj) %>%
  head(cfg$table_max_rows)
```

```{r ric_nutri_volcano_plot, message = FALSE, warning = FALSE}
plot_volcano(deseq_nutri[["ric"]],
             mtorc1_tx_nutri_genes, mtorc2_tx_nutri_genes, mtorc3_tx_nutri_genes,
             "ric",
             "ΔRICTOR Transcriptome Differential Expression (nutrient-fed)",
             TX_XLIMS, TX_YLIMS)
```

### Phosphoproteome (ΔRICTOR nutrient-fed)

Phospho sites with the highest differential abundance between ΔRICTOR and control?

```{r phospho_nutri_ric_top_hits}
phospho_nutri_ric_full %>%
  arrange(padj) %>%
  head(cfg$table_max_rows)
```

How many proteins are associated with multiple phosphorylation sites?

```{r phospho_nutri_ric_multiple_phospho_sites}
table(duplicated(phospho_nutri_ric_full$symbol))
```

```{r ric_nutri_volcano_plot, message = FALSE, warning = FALSE}
plot_volcano(phospho_nutri_ric,
             mtorc1_phospho_nutri_genes, mtorc2_phospho_nutri_genes, mtorc3_phospho_nutri_genes,
             "ric",
             "ΔRICTOR Phosphoproteome Differential Abundance (nutrient-fed)",
             PHOSPHO_XLIMS, PHOSPHO_YLIMS,
             "phospho")
```

### Phosphoproteome vs. transcriptome (ΔRICTOR nutrient-fed)

Are there genes which were found to be differentially expressed in the ΔRICTOR transcriptome data,
which also appear to encode proteins that are differentially phosphorylated?

```{r tx_vs_phospho}
phospho_nutri_ric$tx_pval <- deseq_nutri[["ric"]]$pvalue[match(phospho_nutri_ric$symbol, deseq_nutri[["ric"]]$symbol)]
phospho_nutri_ric$tx_padj <- deseq_nutri[["ric"]]$padj[match(phospho_nutri_ric$symbol, deseq_nutri[["ric"]]$symbol)]

# since there are a large number of significant genes in the tx results, rather than simply looking
# at the intersection of "significant" genes, gene p-value rankings between the tx/phospho data will
# be compared
phospho_nutri_ric$phospho_pval_rank <- rank(phospho_nutri_ric$pval)
phospho_nutri_ric$tx_pval_rank <- rank(phospho_nutri_ric$tx_pval)

phospho_nutri_ric <- phospho_nutri_ric %>%
  mutate(average_pval_rank = (phospho_pval_rank + tx_pval_rank) / 2)
```

Which genes are present and appear significant in both datasets?

```{r ric_nutri_common_sig}
phospho_nutri_ric %>%
  filter(!is.na(tx_pval) & !is.na(symbol)) %>%
  filter(padj <= 0.01 & tx_padj <= 0.01) %>%
  arrange(average_pval_rank) %>%
  head(cfg$table_max_rows)
```

## ΔRICTOR starved

### Transcriptome (ΔRICTOR starved)

Genes with the highest differential expression between ΔRICTOR and control?

```{r tx_starved_ric}
deseq_starved[["ric"]] %>%
  arrange(padj) %>%
  head(cfg$table_max_rows)
```

```{r rictor_starved_volcano_plot, message = FALSE, warning = FALSE}
plot_volcano(deseq_starved[["ric"]],
             mtorc1_tx_starved_genes, mtorc2_tx_starved_genes, mtorc3_tx_starved_genes,
             "ric",
             "ΔRICTOR Transcriptome Differential Expression (starved)",
             TX_XLIMS, TX_YLIMS)
```

### Phosphoproteome (ΔRICTOR starved)

Phospho sites with the highest differential abundance between ΔRICTOR and control?

```{r phospho_starved_ric_top_hits}
phospho_starved_ric_full %>%
  arrange(padj) %>%
  head(cfg$table_max_rows)
```

How many proteins are associated with multiple phosphorylation sites?

```{r phospho_starved_ric_multiple_phospho_sites}
table(duplicated(phospho_starved_ric_full$symbol))
```

```{r rictor_starved_volcano_plot, message = FALSE, warning = FALSE}
# significant genes for each complex knockout
mtorc1_phospho_starved_genes <- phospho_starved_rap %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

mtorc2_phospho_starved_genes <- phospho_starved_ric %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

mtorc3_phospho_starved_genes <- phospho_starved_meak7 %>%
  arrange(padj) %>%
  head(cfg$volcano_num_sig) %>%
  pull(symbol)

plot_volcano(phospho_starved_ric,
             mtorc1_phospho_starved_genes, mtorc2_phospho_starved_genes, mtorc3_phospho_starved_genes,
             "ric",
             "ΔRICTOR Phosphoproteome Differential Abundance (starved)",
             PHOSPHO_XLIMS, PHOSPHO_YLIMS,
             "phospho")
```

### Phosphoproteome vs. transcriptome (ΔRICTOR starved)

Are there genes which were found to be differentially expressed in the ΔRICTOR transcriptome data,
which also appear to encode proteins that are differentially phosphorylated?

```{r tx_vs_phospho}
phospho_starved_ric$tx_pval <- deseq_starved[["ric"]]$pvalue[match(phospho_starved_ric$symbol, deseq_starved[["ric"]]$symbol)]
phospho_starved_ric$tx_padj <- deseq_starved[["ric"]]$padj[match(phospho_starved_ric$symbol, deseq_starved[["ric"]]$symbol)]

# since there are a large number of significant genes in the tx results, rather than simply looking
# at the intersection of "significant" genes, gene p-value rankings between the tx/phospho data will
# be compared
phospho_starved_ric$phospho_pval_rank <- rank(phospho_starved_ric$padj)
phospho_starved_ric$tx_pval_rank <- rank(phospho_starved_ric$tx_pval)

phospho_starved_ric <- phospho_starved_ric %>%
  mutate(average_pval_rank = (phospho_pval_rank + tx_pval_rank) / 2)
```

Which genes are present and appear significant in both datasets?

```{r ric_starved_common_sig}
phospho_starved_ric %>%
  filter(!is.na(tx_pval) & !is.na(symbol)) %>%
  filter(padj <= 0.01 & tx_padj <= 0.01) %>%
  arrange(average_pval_rank) %>%
  head(cfg$table_max_rows)
```

## ΔMEAK7 nutrient-fed

### Transcriptome (ΔMEAK7 nutrient-fed)

Genes with the highest differential expression between ΔMEAK7 and control?

```{r tx_nutri_meak7}
deseq_nutri[["meak7"]] %>%
  arrange(padj) %>%
  head(cfg$table_max_rows)
```

```{r meak7_nutri_volcano_plot, message = FALSE, warning = FALSE}
plot_volcano(deseq_nutri[["meak7"]],
             mtorc1_tx_nutri_genes, mtorc2_tx_nutri_genes, mtorc3_tx_nutri_genes,
             "meak7",
             "ΔMEAK7 Transcriptome Differential Expression (nutrient-fed)",
             TX_XLIMS, TX_YLIMS)
```

### Phosphoproteome (ΔMEAK7 nutrient-fed)

Phospho sites with the highest differential abundance between ΔMEAK7 and control?

```{r phospho_nutri_meak7_top_hits}
phospho_nutri_meak7_full %>%
  arrange(padj) %>%
  head(cfg$table_max_rows)
```

How many proteins are associated with multiple phosphorylation sites?

```{r phospho_nutri_meak7_multiple_phospho_sites}
table(duplicated(phospho_nutri_meak7_full$symbol))
```

```{r meak7_nutri_volcano_plot, message = FALSE, warning = FALSE}
plot_volcano(phospho_nutri_meak7,
             mtorc1_phospho_nutri_genes, mtorc2_phospho_nutri_genes, mtorc3_phospho_nutri_genes,
             "meak7",
             "ΔMEAK7 Phosphoproteome Differential Abundance (nutrient-fed)",
             PHOSPHO_XLIMS, PHOSPHO_YLIMS,
             "phospho")
```

### Phosphoproteome vs. transcriptome (ΔMEAK7 nutrient-fed)

Are there genes which were found to be differentially expressed in the ΔMEAK7 transcriptome data,
which also appear to encode proteins that are differentially phosphorylated?

```{r tx_vs_phospho}
phospho_nutri_meak7$tx_pval <- deseq_nutri[["meak7"]]$pvalue[match(phospho_nutri_meak7$symbol, deseq_nutri[["meak7"]]$symbol)]
phospho_nutri_meak7$tx_padj <- deseq_nutri[["meak7"]]$padj[match(phospho_nutri_meak7$symbol, deseq_nutri[["meak7"]]$symbol)]

# since there are a large number of significant genes in the tx results, rather than simply looking
# at the intersection of "significant" genes, gene p-value rankings between the tx/phospho data will
# be compared
phospho_nutri_meak7$phospho_pval_rank <- rank(phospho_nutri_meak7$pval)
phospho_nutri_meak7$tx_pval_rank <- rank(phospho_nutri_meak7$tx_pval)

phospho_nutri_meak7 <- phospho_nutri_meak7 %>%
  mutate(average_pval_rank = (phospho_pval_rank + tx_pval_rank) / 2)
```

Which genes are present and appear significant in both datasets?

```{r meak7_nutri_common_sig}
phospho_nutri_meak7 %>%
  filter(!is.na(tx_pval) & !is.na(symbol)) %>%
  filter(padj <= 0.01 & tx_padj <= 0.01) %>%
  arrange(average_pval_rank) %>%
  head(cfg$table_max_rows)
```

## ΔMEAK7 starved

### Transcriptome (ΔMEAK7 starved)

Genes with the highest differential expression between ΔMEAK7 and control?

```{r tx_starved_meak7}
deseq_starved[["meak7"]] %>%
  arrange(padj) %>%
  head(cfg$table_max_rows)
```

```{r rictor_starved_volcano_plot, message = FALSE, warning = FALSE}
plot_volcano(deseq_starved[["meak7"]],
             mtorc1_tx_starved_genes, mtorc2_tx_starved_genes, mtorc3_tx_starved_genes,
             "meak7",
             "ΔMEAK7 Transcriptome Differential Expression (starved)",
             TX_XLIMS, TX_YLIMS)
```

### Phosphoproteome (ΔMEAK7 starved)

Phospho sites with the highest differential abundance between ΔMEAK7 and control?

```{r phospho_starved_meak7_top_hits}
phospho_starved_meak7_full %>%
  arrange(padj) %>%
  head(cfg$table_max_rows)
```

How many proteins are associated with multiple phosphorylation sites?

```{r phospho_starved_meak7_multiple_phospho_sites}
table(duplicated(phospho_starved_meak7_full$symbol))
```

```{r meak7_starved_volcano_plot, message = FALSE, warning = FALSE}
plot_volcano(phospho_starved_meak7,
             mtorc1_phospho_starved_genes, mtorc2_phospho_starved_genes, mtorc3_phospho_starved_genes,
             "meak7",
             "ΔMEAK7 Phosphoproteome Differential Abundance (starved)",
             PHOSPHO_XLIMS, PHOSPHO_YLIMS,
             "phospho")
```

### Phosphoproteome vs. transcriptome (ΔMEAK7 starved)

Are there genes which were found to be differentially expressed in the ΔMEAK7 transcriptome data,
which also appear to encode proteins that are differentially phosphorylated?

```{r tx_vs_phospho}
phospho_starved_meak7$tx_pval <- deseq_starved[["meak7"]]$pvalue[match(phospho_starved_meak7$symbol, deseq_starved[["meak7"]]$symbol)]
phospho_starved_meak7$tx_padj <- deseq_starved[["meak7"]]$padj[match(phospho_starved_meak7$symbol, deseq_starved[["meak7"]]$symbol)]

# since there are a large number of significant genes in the tx results, rather than simply looking
# at the intersection of "significant" genes, gene p-value rankings between the tx/phospho data will
# be compared
phospho_starved_meak7$phospho_pval_rank <- rank(phospho_starved_meak7$padj)
phospho_starved_meak7$tx_pval_rank <- rank(phospho_starved_meak7$tx_pval)

phospho_starved_meak7 <- phospho_starved_meak7 %>%
  mutate(average_pval_rank = (phospho_pval_rank + tx_pval_rank) / 2)
```

Which genes are present and appear significant in both datasets?

```{r meak7_starved_common_sig}
phospho_starved_meak7 %>%
  filter(!is.na(tx_pval) & !is.na(symbol)) %>%
  filter(padj <= 0.01 & tx_padj <= 0.01) %>%
  arrange(average_pval_rank) %>%
  head(cfg$table_max_rows)
```

## Commonly affected genes

Nutrient-fed:

```{r tx_nutri_common_genes}
deseq_nutri[["combined"]] %>%
  head(cfg$table_max_rows)
```

Starved:

```{r tx_starved_common_genes}
deseq_starved[["combined"]] %>%
  head(cfg$table_max_rows)
```

## Commonly affected phosphorylation sites

Nutrient-fed:

```{r phospho_nutri_common_genes}
phospho_nutri %>%
  arrange(padj_all) %>%
  head(cfg$table_max_rows) %>%
  kable(digits = 3)
```

Starved:

```{r phospho_starved_common_genes}
phospho_starved %>%
  arrange(padj_all) %>%
  head(cfg$table_max_rows) %>%
  kable(digits = 3)
```

## Significant gene overlap Venn diagrams

Below, Venn diagrams are created depicting the overlap in statistically significant genes from the
differential expression/abundance tests for each of the experiments, considering up- and
down-regulated genes separately.

As a reminder, the phosphoproteomics experiments measured differences at the levels of phosphosites
within proteins. Here, the "genes" referred to for the phosphoproteomics results are the inferred
genes encoding those proteins.

```{r venn_helper_func}
# helper function to generate venn diagrams
plot_venn <- function(rap, ric, m7, plot_title) {
  # colors to use
  venn_colors <- c("#E7EECE", "#EEC8A1", "#EDC8C6")
  border_colors <- c("#C2D18D", "#CD9155", "#D07C78")

  venn_labels <- sprintf(c("Rap KO(n=%d)", "m7 KO (n=%d)", "Ric KO (n=%d)"),
                         c(length(rap), length(m7), length(ric)))

  # plot venn diagram
  # https://r-graph-gallery.com/14-venn-diagramm.html
  plt <- venn.diagram(
    x = list(rap, m7, ric),
    category.names = venn_labels,
    output = TRUE,
    filename = NULL,

    imagetype = "png",
    height = 960,
    width = 960,
    resolution = 192,
    compression = "lzw",

    lwd = 1,
    lty = "solid",
    fill = venn_colors,
    col = border_colors,
    rotation = 1,

    cex = .6,
    fontface = "bold",
    fontfamily = "sans",
    print.mode = c("raw", "percent"),

    cat.cex = 0.6,
    cat.fontface = "bold",
    cat.default.pos = "outer",
    cat.pos = c(-27, 27, 135),
    cat.dist = c(0.055, 0.055, 0.085),
    cat.fontfamily = "sans",

    main = plot_title
  )

  # display plot
  grid::grid.draw(plt)
}
```

```{r venn_gene_sets}
# get lists of significant genes for each experiment, separated by up-/down-regulation
tx_starved_rap_up <- deseq_starved$rap %>%
  filter(padj < 0.01 & log2FoldChange > 0) %>%
  pull(ensgene)

tx_starved_ric_up <- deseq_starved$ric %>%
  filter(padj < 0.01 & log2FoldChange > 0) %>%
  pull(ensgene)

tx_starved_m7_up <- deseq_starved$meak7 %>%
  filter(padj < 0.01 & log2FoldChange > 0) %>%
  pull(ensgene)

tx_starved_rap_down <- deseq_starved$rap %>%
  filter(padj < 0.01 & log2FoldChange < 0) %>%
  pull(ensgene)

tx_starved_ric_down <- deseq_starved$ric %>%
  filter(padj < 0.01 & log2FoldChange < 0) %>%
  pull(ensgene)

tx_starved_m7_down <- deseq_starved$meak7 %>%
  filter(padj < 0.01 & log2FoldChange < 0) %>%
  pull(ensgene)

tx_nutri_rap_up <- deseq_nutri$rap %>%
  filter(padj < 0.01 & log2FoldChange > 0) %>%
  pull(ensgene)

tx_nutri_ric_up <- deseq_nutri$ric %>%
  filter(padj < 0.01 & log2FoldChange > 0) %>%
  pull(ensgene)

tx_nutri_m7_up <- deseq_nutri$meak7 %>%
  filter(padj < 0.01 & log2FoldChange > 0) %>%
  pull(ensgene)

tx_nutri_rap_down <- deseq_nutri$rap %>%
  filter(padj < 0.01 & log2FoldChange < 0) %>%
  pull(ensgene)

tx_nutri_ric_down <- deseq_nutri$ric %>%
  filter(padj < 0.01 & log2FoldChange < 0) %>%
  pull(ensgene)

tx_nutri_m7_down <- deseq_nutri$meak7 %>%
  filter(padj < 0.01 & log2FoldChange < 0) %>%
  pull(ensgene)

# phospho results; here, a less stringent p-value cutoff is used to account for the lower
# statistical power of the data compared with the transcriptomics datasets
phospho_starved_rap_up <- phospho_starved %>%
  filter(rap_padj < 0.05 & rap_log2ratio > 0) %>%
  drop_na(symbol) %>%
  pull(symbol)

phospho_starved_ric_up <- phospho_starved %>%
  filter(ric_padj < 0.05 & ric_log2ratio > 0) %>%
  drop_na(symbol) %>%
  pull(symbol)

phospho_starved_meak7_up <- phospho_starved %>%
  filter(meak7_padj < 0.05 & meak7_log2ratio > 0) %>%
  drop_na(symbol) %>%
  pull(symbol)

phospho_starved_rap_down <- phospho_starved %>%
  filter(rap_padj < 0.05 & rap_log2ratio < 0) %>%
  drop_na(symbol) %>%
  pull(symbol)

phospho_starved_ric_down <- phospho_starved %>%
  filter(ric_padj < 0.05 & ric_log2ratio < 0) %>%
  drop_na(symbol) %>%
  pull(symbol)

phospho_starved_meak7_down <- phospho_starved %>%
  filter(meak7_padj < 0.05 & meak7_log2ratio < 0) %>%
  drop_na(symbol) %>%
  pull(symbol)

phospho_nutri_rap_up <- phospho_nutri %>%
  filter(rap_padj < 0.05 & rap_log2ratio > 0) %>%
  drop_na(symbol) %>%
  pull(symbol)

phospho_nutri_ric_up <- phospho_nutri %>%
  filter(ric_padj < 0.05 & ric_log2ratio > 0) %>%
  drop_na(symbol) %>%
  pull(symbol)

phospho_nutri_meak7_up <- phospho_nutri %>%
  filter(meak7_padj < 0.05 & meak7_log2ratio > 0) %>%
  drop_na(symbol) %>%
  pull(symbol)

phospho_nutri_rap_down <- phospho_nutri %>%
  filter(rap_padj < 0.05 & rap_log2ratio < 0) %>%
  drop_na(symbol) %>%
  pull(symbol)

phospho_nutri_ric_down <- phospho_nutri %>%
  filter(ric_padj < 0.05 & ric_log2ratio < 0) %>%
  drop_na(symbol) %>%
  pull(symbol)

phospho_nutri_meak7_down <- phospho_nutri %>%
  filter(meak7_padj < 0.05 & meak7_log2ratio < 0) %>%
  drop_na(symbol) %>%
  pull(symbol)
```

```{r tx_starved_down_venn, fig.height = 6}
plot_venn(tx_starved_rap_down, tx_starved_ric_down, tx_starved_m7_down,
          "Transcriptomics (starved / down-regulated)")
```

```{r tx_starved_up_venn, fig.height = 6}
plot_venn(tx_starved_rap_up, tx_starved_ric_up, tx_starved_m7_up,
          "Transcriptomics (starved / up-regulated)")
```

```{r tx_nutri_down_venn, fig.height = 6}
plot_venn(tx_nutri_rap_down, tx_nutri_ric_down, tx_nutri_m7_down,
          "Transcriptomics (nutrient-fed / down-regulated)")
```

```{r tx_nutri_up_venn, fig.height = 6}
plot_venn(tx_nutri_rap_up, tx_nutri_ric_up, tx_nutri_m7_up,
          "Transcriptomics (nutrient-fed / up-regulated)")
```

```{r phospho_starved_down_venn, fig.height = 6}
plot_venn(phospho_starved_rap_down, phospho_starved_ric_down, phospho_starved_meak7_down,
          "Phosphoproteomics (starved / down-regulated)")
```

```{r phospho_starved_up_venn, fig.height = 6}
plot_venn(phospho_starved_rap_up, phospho_starved_ric_up, phospho_starved_meak7_up,
          "Phosphoproteomics (starved / up-regulated)")
```

```{r phospho_nutri_down_venn, fig.height = 6}
plot_venn(phospho_nutri_rap_down, phospho_nutri_ric_down, phospho_nutri_meak7_down,
          "Phosphoproteomics (nutrient-fed / down-regulated)")
```

```{r phospho_nutri_up_venn, fig.height = 6}
plot_venn(phospho_nutri_rap_up, phospho_nutri_ric_up, phospho_nutri_meak7_up,
          "Phosphoproteomics (nutrient-fed / up-regulated)")
```

## Save Results

```{r write_output}
# Phosphosite-level Proteasome Discoverer
phospho_tbls <- list(
  "RPTOR (nutrient-fed)" = phospho_nutri_rap_full,
  "RICTOR (nutrient-fed)" = phospho_nutri_ric_full,
  "MEAK7 (nutrient-fed)" = phospho_nutri_meak7_full,
  "RPTOR (starved)" = phospho_starved_rap_full,
  "RICTOR (starved)" = phospho_starved_ric_full,
  "MEAK7 (starved)" = phospho_starved_meak7_full
)

phospho_full_outfile <- file.path(cfg$output_dir, "proteome_discoverer_phosphosites_all.xlsx")
write.xlsx(phospho_tbls, file = phospho_full_outfile)

# Subset of Phosphosite-level Proteasome Discoverer with the most significant entry for each protein
phospho_subset_tbls <- list(
  "RPTOR (nutrient-fed)" = phospho_nutri_rap,
  "RICTOR (nutrient-fed)" = phospho_nutri_ric,
  "MEAK7 (nutrient-fed)" = phospho_nutri_meak7,
  "RPTOR (starved)" = phospho_starved_rap,
  "RICTOR (starved)" = phospho_starved_ric,
  "MEAK7 (starved)" = phospho_starved_meak7
)

phospho_outfile <- file.path(cfg$output_dir, "proteome_discoverer_phosphosites_top.xlsx")
write.xlsx(phospho_subset_tbls, file = phospho_outfile)

# DESeq2 results
deseq_tbls = list(
  "RPTOR (nutrient-fed)" = deseq_nutri[["rap"]],
  "RICTOR (nutrient-fed)" = deseq_nutri[["ric"]],
  "MEAK7 (nutrient-fed)" = deseq_nutri[["meak7"]],
  "Combined (nutrient-fed)" = deseq_nutri[["combined"]],
  "RPTOR (starved)" = deseq_starved[["rap"]],
  "RICTOR (starved)" = deseq_starved[["ric"]],
  "MEAK7 (starved)" = deseq_starved[["meak7"]],
  "Combined (starved)" = deseq_starved[["combined"]]
)

tx_outfile <- file.path(cfg$output_dir, "deseq2_results.xlsx")
write.xlsx(deseq_tbls, file = tx_outfile)
```

```{r bev_phospho_nutri_unique_sig}
# answering a request from Bev to provide lists of the unique phospho/nutri genes associated with
# each complex (grouping up- and down-regulated genes together)
rap_sig <- sort(unique(c(phospho_nutri_rap_up, phospho_nutri_rap_down)))
ric_sig <- sort(unique(c(phospho_nutri_ric_up, phospho_nutri_ric_down)))
meak7_sig <- sort(unique(c(phospho_nutri_meak7_up, phospho_nutri_meak7_down)))

rap_unique <- rap_sig[!rap_sig %in% c(ric_sig, meak7_sig)]
ric_unique <- ric_sig[!ric_sig %in% c(rap_sig, meak7_sig)]
meak7_unique <- meak7_sig[!meak7_sig %in% c(rap_sig, ric_sig)]

df1 <- phospho_nutri_rap %>%
  filter(symbol %in% rap_unique) %>%
  select(symbol, log2ratio, pval, padj) %>%
  arrange(padj)

df2 <- phospho_nutri_ric %>%
  filter(symbol %in% ric_unique) %>%
  select(symbol, log2ratio, pval, padj) %>%
  arrange(padj)

df3 <- phospho_nutri_meak7 %>%
  filter(symbol %in% meak7_unique) %>%
  select(symbol, log2ratio, pval, padj) %>%
  arrange(padj)

write_tsv(df1, "~/rap_unique_phospho.tsv")
write_tsv(df2, "~/ric_unique_phospho.tsv")
write_tsv(df3, "~/meak7_unique_phospho.tsv")
```

```{r bev_tx_nutri_unique_sig}
rap_sig <- sort(unique(c(tx_nutri_rap_up, tx_nutri_rap_down)))
ric_sig <- sort(unique(c(tx_nutri_ric_up, tx_nutri_ric_down)))
meak7_sig <- sort(unique(c(tx_nutri_m7_up, tx_nutri_m7_down)))

rap_unique <- rap_sig[!rap_sig %in% c(ric_sig, meak7_sig)]
ric_unique <- ric_sig[!ric_sig %in% c(rap_sig, meak7_sig)]
meak7_unique <- meak7_sig[!meak7_sig %in% c(rap_sig, ric_sig)]

df1 <- deseq_nutri[["rap"]] %>%
  filter(ensgene %in% rap_unique) %>%
  select(symbol, log2FoldChange, pvalue, padj) %>%
  arrange(padj)

df2 <- deseq_nutri[["ric"]] %>%
  filter(ensgene %in% ric_unique) %>%
  select(symbol, log2FoldChange, pvalue, padj) %>%
  arrange(padj)

df3 <- deseq_nutri[["meak7"]] %>%
  filter(ensgene %in% meak7_unique) %>%
  select(symbol, log2FoldChange, pvalue, padj) %>%
  arrange(padj)

write_tsv(df1, "~/rap_unique_tx.tsv")
write_tsv(df2, "~/ric_unique_tx.tsv")
write_tsv(df3, "~/meak7_unique_tx.tsv")
```

# Session Info

```{r session_info}
sessionInfo()
```

Input data MD5 sums:

```{r file_info, results="asis"}
input_files <- c(phospho_nutri_infile,
                 phospho_starved_infile,
                 cfg$uniprot_mapping,
                 file.path(deseq_dir, "condition_Rap_nutri_vs_v_nutri.tsv"),
                 file.path(deseq_dir, "condition_Ric_nutri_vs_v_nutri.tsv"),
                 file.path(deseq_dir, "condition_m7_nutri_vs_v_nutri.tsv"),
                 file.path(deseq_dir, "condition_Rap_starved_vs_v_starved.tsv"),
                 file.path(deseq_dir, "condition_Ric_starved_vs_v_starved.tsv"),
                 file.path(deseq_dir, "condition_m7_starved_vs_v_starved.tsv"))

out <- ""

for (file in input_files) {
  out <- paste0(out, sprintf("- **%s**: %s\n", basename(file), tools::md5sum(file)))
}

cat(out)
```
