---
title: "CRISPR-guided multiomics unveils direct mTOR inhibition of Hippo signaling through CDCP1: Phosphoproteomics GSEA Analysis"
output:
  html_document:
    df_print: kable
  pdf_document:
    df_print: kable
---

## Overview

Here, gene set enrichment analysis (GSEA) is performed for each of the three mTOR
complex knockouts, with genes weighted and ranked based on the ratio of average gene
expression for the knockouts compared to controls.

This analysis was performed separately for nutrient-fed and starved samples, and was
performed for both phosphoproteomics data (here) and RNA-Seq data (separate file.)

To perform the actual gene set enrichment analysis, the `GSEA()` function from the
[clusterProfiler](https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html) 
Bioconductor package is used.

`GSEA()` provides a helpful wrapper around [Fast Gene Set Enrichment Analysis
(fgsea)](https://bioconductor.org/packages/release/bioc/html/fgsea.html), and makes it
easy to visualize the results of enrichment analyses.

All of the code used to perform this analysis, along with the resulting tables and
figures depicting significantly enriched gene sets are shown in this [RMarkdown](https://rmarkdown.rstudio.com/) 
document.

The full source .Rmd as well as the input datasets used to perform the analysis can be
found at: [https://github.com/khughitt/nguyen22-crispr-guided-multiomics](https://github.com/khughitt/nguyen22-crispr-guided-multiomics)

## Phosphoproteomics vs. Transcriptomics analysis

In general, the same analysis approach was used to assess gene set enrichment for both
the transcriptomic (RNA-Seq) and phosphoproteomic samples.

With respect to the present analysis, a major different between the two data types,
however, is the smaller number of genes for phospho abundance could be quantified.

Whereas the transcriptomics dataset includes measurements for ~61,500 genes, the phospho
dataset includes measurements for only ~3,200 genes.
This is due both to the different natures of the assays, as well as to the fact that the
phosphoproteomic data is measured at the _protein_ level, and not all proteins are able
to be mapped to gene symbols.

The effect of this is that there are far fewer genes in the universe of query genes used
for the phosphoproteomics enrichment analysis, compared with the transcriptomics
enrichment analysis, and thus, a more limited ability to detect significantly enrichmed
gene sets for this data.

In order to show potentially interesting / "trending" gene sets, a more relaxed adjusted
P-value cutoff of "0.15" was used for the phosphoprotetomics analysis.

For a more detailed description of each of the major steps in the analysis, refer to the
_nutrient-fed_ section of the _transcriptomics_ gene set enrichment results.

## Setup

```{r message = FALSE}
library(clusterProfiler)
library(enrichplot)
library(ggnewscale)
library(msigdbr)
library(tidyverse)
library(xlsx)

# plot opts
knitr::opts_chunk$set(
  fig.width = 6,
  fig.height = 3.6,
  fig.retina = 1,
  dpi = 192
)

# output directory?
out_dir <- "output/"

FGSEA_NPERM <- 10000
MIN_PADJ <- 0.15
```

```{r load_gene_sets}
# load gene MSIGdb gene sets
gene_sets <- msigdbr(category = "H")

gene_sets <- gene_sets %>%
  select(gs_name, gene_symbol) %>%
  distinct()
```

## Nutrient-fed

```{r load_nutri_data, message=FALSE}
nutri <- read_csv('data/phosphoproteomics_nutri.csv', show_col_types = FALSE)[, -1]
nutri$symbol <- str_split(nutri$Phospho_Site, " ", simplify=TRUE)[, 1]

# drop small number of entries with missing p-values
# nutri <- nutri[complete.cases(nutri), ]

# perform size factor normalization
ind <- grepl("_[1-4]$", colnames(nutri))
nutri[, ind] <- sweep(nutri[, ind], 2, colSums(nutri[, ind]), '/') * 1E6

dim(nutri)

# compute condition-specific totals
nutri_rap <- nutri %>%
  select(starts_with("Rap")) %>%
  rowMeans()

nutri_ric <- nutri %>%
  select(starts_with("Ric")) %>%
  rowMeans()

nutri_m7 <- nutri %>%
  select(starts_with("m7")) %>%
  rowMeans()

nutri_control <- nutri %>%
  select(starts_with("control")) %>%
  rowMeans()

# compute knockout vs. control ratios;
nutri$Ratio_Rap_v_Control <- (nutri_rap + 1) / (nutri_control + 1)
nutri$Ratio_Ric_v_Control <- (nutri_ric + 1) / (nutri_control + 1)
nutri$Ratio_m7_v_Control <- (nutri_m7 + 1) / (nutri_control + 1)
```

```{r}
# preview data
nutri[1:5, 1:5]
```

```{r create_gene_lists_nutri}
# construct gene score vectors for GSEA analysis;
# in cases where multiple phospho sites or proteins map to the same gene, keep the
# ratio with the largest absolute deviance from "0"
mtorc1_nutri_genes <- nutri %>%
  select(symbol, ratio = Ratio_Rap_v_Control) %>%
  mutate(log2ratio = log2(ratio)) %>%
  mutate(log2ratio_abs = abs(log2ratio)) %>%
  group_by(symbol) %>%
  arrange(-log2ratio_abs) %>%
  slice(1) %>%
  ungroup() %>%
  select(symbol, log2ratio) %>%
  arrange(-log2ratio) %>%
  deframe()

mtorc2_nutri_genes <- nutri %>%
  select(symbol, ratio = Ratio_Ric_v_Control) %>%
  mutate(log2ratio = log2(ratio)) %>%
  mutate(log2ratio_abs = abs(log2ratio)) %>%
  group_by(symbol) %>%
  arrange(-log2ratio_abs) %>%
  slice(1) %>%
  ungroup() %>%
  select(symbol, log2ratio) %>%
  arrange(-log2ratio) %>%
  deframe()

mtorc3_nutri_genes <- nutri %>%
  select(symbol, ratio = Ratio_m7_v_Control) %>%
  mutate(log2ratio = log2(ratio)) %>%
  mutate(log2ratio_abs = abs(log2ratio)) %>%
  group_by(symbol) %>%
  arrange(-log2ratio_abs) %>%
  slice(1) %>%
  ungroup() %>%
  select(symbol, log2ratio) %>%
  arrange(-log2ratio) %>%
  deframe()
```

```{r}
head(enframe(mtorc1_nutri_genes))
head(enframe(mtorc2_nutri_genes))
head(enframe(mtorc3_nutri_genes))
```

```{r fgsea, message = FALSE, warning = FALSE}
mtorc1_nutri_res <- GSEA(mtorc1_nutri_genes, TERM2GENE = gene_sets, seed = 321, pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
mtorc2_nutri_res <- GSEA(mtorc2_nutri_genes, TERM2GENE = gene_sets, seed = 321, pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
mtorc3_nutri_res <- GSEA(mtorc3_nutri_genes, TERM2GENE = gene_sets, seed = 321, pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
```

```{r mtorc1_results}
mtorc1_nutri_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

dotplot(mtorc1_nutri_res) +
  ggplot2::xlim(0, 1.1)
```

```{r mtorc2_nutri_results}
mtorc2_nutri_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

dotplot(mtorc2_nutri_res) +
  ggplot2::xlim(0, 1.1)
```

```{r mtorc3_nutri_results}
mtorc3_nutri_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

dotplot(mtorc3_nutri_res) +
  ggplot2::xlim(0, 1.1)
```

## Starved

```{r load_starved_data, message=FALSE}
starved <- read_csv('data/phosphoproteomics_starved.csv', show_col_types = FALSE)[, -1]
starved$symbol <- str_split(starved$Phospho_Site, " ", simplify=TRUE)[, 1]
dim(starved)

# size factor normalization
ind <- grepl("_[1-4]$", colnames(starved))
starved[, ind] <- sweep(starved[, ind], 2, colSums(starved[, ind]), '/') * 1E6

dim(starved)

# compute condition-specific totals
starved_rap <- starved %>%
  select(starts_with("Rap")) %>%
  rowSums()

starved_ric <- starved %>%
  select(starts_with("Ric")) %>%
  rowSums()

starved_m7 <- starved %>%
  select(starts_with("m7")) %>%
  rowSums()

starved_control <- starved %>%
  select(starts_with("control")) %>%
  rowSums()

# compute knockout vs. control ratios
starved$Ratio_Rap_v_Control <- (starved_rap + 1) / (starved_control + 1)
starved$Ratio_Ric_v_Control <- (starved_ric + 1) / (starved_control + 1)
starved$Ratio_m7_v_Control  <- (starved_m7 +1 ) / (starved_control + 1)
```

```{r}
# preview data
starved[1:5, 1:5]
```

```{r create_gene_lists_starved}
# construct gene score vectors for GSEA analysis;
# in cases where multiple phospho sites or proteins map to the same gene, keep the
# ratio with the largest absolute deviance from "0"
mtorc1_starved_genes <- starved %>%
  select(symbol, ratio = Ratio_Rap_v_Control) %>%
  mutate(log2ratio = log2(ratio)) %>%
  mutate(log2ratio_abs = abs(log2ratio)) %>%
  group_by(symbol) %>%
  arrange(-log2ratio_abs) %>%
  slice(1) %>%
  ungroup() %>%
  select(symbol, log2ratio) %>%
  arrange(-log2ratio) %>%
  deframe()

mtorc2_starved_genes <- starved %>%
  select(symbol, ratio = Ratio_Ric_v_Control) %>%
  mutate(log2ratio = log2(ratio)) %>%
  mutate(log2ratio_abs = abs(log2ratio)) %>%
  group_by(symbol) %>%
  arrange(-log2ratio_abs) %>%
  slice(1) %>%
  ungroup() %>%
  select(symbol, log2ratio) %>%
  arrange(-log2ratio) %>%
  deframe()

mtorc3_starved_genes <- starved %>%
  select(symbol, ratio = Ratio_m7_v_Control) %>%
  mutate(log2ratio = log2(ratio)) %>%
  mutate(log2ratio_abs = abs(log2ratio)) %>%
  group_by(symbol) %>%
  arrange(-log2ratio_abs) %>%
  slice(1) %>%
  ungroup() %>%
  select(symbol, log2ratio) %>%
  arrange(-log2ratio) %>%
  deframe()
```

```{r}
head(enframe(mtorc1_starved_genes))
head(enframe(mtorc2_starved_genes))
head(enframe(mtorc3_starved_genes))
```

```{r fgsea_starved, message = FALSE, warning = FALSE}
mtorc1_starved_res <- GSEA(mtorc1_starved_genes, TERM2GENE = gene_sets, seed = 321, pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
mtorc2_starved_res <- GSEA(mtorc2_starved_genes, TERM2GENE = gene_sets, seed = 321, pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
mtorc3_starved_res <- GSEA(mtorc3_starved_genes, TERM2GENE = gene_sets, seed = 321, pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
```

```{r mtorc1_starved_results}
mtorc1_starved_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

dotplot(mtorc1_starved_res) +
  ggplot2::xlim(0, 1.1)
```

```{r mtorc2_starved_results}
mtorc2_starved_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

dotplot(mtorc2_starved_res) +
  ggplot2::xlim(0, 1.1)
```

```{r mtorc3_starved_results}
mtorc3_starved_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

dotplot(mtorc3_starved_res) +
  ggplot2::xlim(0, 1.1)
```

## Save results

```{r save_results}
if (!dir.exists(out_dir)) {
  dir.create(out_dir, mode = "0755")
}

# save individual tsv files
write_tsv(mtorc1_nutri_res@result, file.path(out_dir, "phosphoproteomics-gsea-results-nutri_mtorc1.tsv"))
write_tsv(mtorc2_nutri_res@result, file.path(out_dir, "phosphoproteomics-gsea-results-nutri_mtorc2.tsv"))
write_tsv(mtorc3_nutri_res@result, file.path(out_dir, "phosphoproteomics-gsea-results-nutri_mtorc3.tsv"))

write_tsv(mtorc1_starved_res@result, file.path(out_dir, "phosphoproteomics-gsea-results-starved_mtorc1.tsv"))
write_tsv(mtorc2_starved_res@result, file.path(out_dir, "phosphoproteomics-gsea-results-starved_mtorc2.tsv"))
write_tsv(mtorc3_starved_res@result, file.path(out_dir, "phosphoproteomics-gsea-results-starved_mtorc3.tsv"))

# save combined xls file
xlsx_outfile <- file.path(out_dir, "phosphoproteomics-gsea-results.xlsx")

write.xlsx(mtorc1_nutri_res@result, file = xlsx_outfile, sheetName = "mTORC1-nutri")
write.xlsx(mtorc1_starved_res@result, file = xlsx_outfile, sheetName = "mTORC1-starved", append = TRUE)
write.xlsx(mtorc2_nutri_res@result, file = xlsx_outfile, sheetName = "mTORC2-nutri", append = TRUE)
write.xlsx(mtorc2_starved_res@result, file = xlsx_outfile, sheetName = "mTORC2-starved", append = TRUE)
write.xlsx(mtorc3_nutri_res@result, file = xlsx_outfile, sheetName = "mTORC3-nutri", append = TRUE)
write.xlsx(mtorc3_starved_res@result, file = xlsx_outfile, sheetName = "mTORC3-starved", append = TRUE)
```

## Session Info

```{r}
sessionInfo()
```
