---
title: "CRISPR-guided multiomics unveils direct mTOR inhibition of Hippo signaling through CDCP1: Phosphoproteomics GSEA Analysis"
output:
  html_document:
    df_print: kable
  pdf_document:
    df_print: kable
---

## Overview

Here, gene set enrichment analysis (GSEA) is performed for each of the three mTOR
complex knockouts, for the set of phospho sites quantified which could be mapped back to
protein-coding genes.

This analysis was performed separately for nutrient-fed and starved samples, and was
performed for both phosphoproteomics data (here) and RNA-Seq data (separate file.)

To perform the actual gene set enrichment analysis, the `GSEA()` function from the
[clusterProfiler](https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html)
Bioconductor package is used.

`GSEA()` provides a helpful wrapper around [Fast Gene Set Enrichment Analysis
(fgsea)](https://bioconductor.org/packages/release/bioc/html/fgsea.html), and makes it
easy to visualize the results of enrichment analyses.

All of the code used to perform this analysis, along with the resulting tables and
figures depicting significantly enriched gene sets are shown in this [RMarkdown](https://rmarkdown.rstudio.com/)
document.

The full source .Rmd as well as the input datasets used to perform the analysis can be
found at: [https://github.com/khughitt/nguyen22-crispr-guided-multiomics](https://github.com/khughitt/nguyen22-crispr-guided-multiomics)

## Phosphoproteomics vs. Transcriptomics analysis

In general, the same analysis approach was used to assess gene set enrichment for both
the transcriptomic (RNA-Seq) and phosphoproteomic samples.

With respect to the present analysis, a major different between the two data types,
however, is the smaller number of genes for phospho abundance could be quantified.

Whereas the transcriptomics dataset includes measurements for ~61,500 genes, the phospho datasets
contain measurements for only ~1,450 proteins that can be reliably mapped to genes.

This is due both to the different natures of the assays, as well as to the fact that the
phosphoproteomic data is measured at the _protein_ level, and not all proteins are able
to be mapped to gene symbols.

The effect of this is that there are far fewer genes in the universe of query genes used
for the phosphoproteomics enrichment analysis, compared with the transcriptomics
enrichment analysis, and thus, a more limited ability to detect significantly enrichmed
gene sets for this data.

In order to show potentially interesting / "trending" gene sets, a more relaxed adjusted
P-value cutoff of "0.15" was used for the phosphoprotetomics analysis.

For a more detailed description of each of the major steps in the analysis, refer to the
_nutrient-fed_ section of the _transcriptomics_ gene set enrichment results.

## Setup

```{r message = FALSE}
library(clusterProfiler)
library(enrichplot)
library(ggnewscale)
library(msigdbr)
library(tidyverse)
library(xlsx)

# plot opts
knitr::opts_chunk$set(
  fig.width = 6,
  fig.height = 3.6,
  fig.retina = 1,
  dpi = 192
)

# output directory?
out_dir <- "output/"

# path to uniprot id mapping file
# https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/idmapping/by_organism/
# version used: 2023-03-01
uniprot_mapping <- "/data/ref/uniprot/2023_03/HUMAN_9606_idmapping.dat.gz"

FGSEA_NPERM <- 10000
MIN_PADJ <- 0.15
```

```{r load_uniprot_id_mapping}
uniprot <- read_tsv(uniprot_mapping, col_names = c("uniprotkb", "id_type", "id")) %>%
  filter(id_type == "Gene_Name") %>%
  select(-id_type)
```

```{r load_gene_sets}
# load gene MSIGdb gene sets
gene_sets <- msigdbr(category = "H")

gene_sets <- gene_sets %>%
  select(gs_name, gene_symbol) %>%
  distinct()
```

```{r phospho_helper_func}
# helper function to prepare phospho data
load_phospho_data <- function(path) {
  dat <- read_tsv(path, show_col_types = FALSE) 

  # add log2 abundance ratios, if missing
  if (!"Abundance Ratio (log2): (Rap_del_1R) / (V_del_4R)" %in% colnames(dat)) {
    dat$`Abundance Ratio (log2): (Rap_del_1R) / (V_del_4R)` = log2(dat$`Abundance Ratio: (Rap_del_1R) / (V_del_4R)`)
    dat$`Abundance Ratio (log2): (Ric_del_SR) / (V_del_4R)` = log2(dat$`Abundance Ratio: (Ric_del_SR) / (V_del_4R)`)
    dat$`Abundance Ratio (log2): (M7_del_SR) / (V_del_4R)` = log2(dat$`Abundance Ratio: (M7_del_SR) / (V_del_4R)`)
  }

  dat <- dat %>%
    select(
      modifications = Modifications,
      master_protein_accessions = `Master Protein Accessions`,
      rap_vs_control_log2ratio = `Abundance Ratio (log2): (Rap_del_1R) / (V_del_4R)`,
      ric_vs_control_log2ratio = `Abundance Ratio (log2): (Ric_del_SR) / (V_del_4R)`,
      m7_vs_control_log2ratio = `Abundance Ratio (log2): (M7_del_SR) / (V_del_4R)`
    )

  # exclude entries that doesn't relate to phosphorylation events
  mask <- grepl("Phospho", dat$modifications)
  dat <- dat[mask, ]

  # exclude entries mapping to multiple proteins
  mask <- !grepl(";", dat$master_protein_accessions)
  dat <- dat[mask, ]

  # exclude entries with ids not found in the UniprotKB mapping file
  mask <- dat$master_protein_accessions %in% uniprot$uniprotkb
  dat <- dat[mask, ]

  # add gene symbol column
  dat$symbol <- uniprot$id[match(dat$master_protein_accessions, uniprot$uniprotkb)]

  # drop fields that are no longer needed and move symbols to front
  dat <- dat %>%
    select(symbol, rap_vs_control_log2ratio, ric_vs_control_log2ratio, m7_vs_control_log2ratio)

  # drop entries with missing p-values
  dat <- dat[complete.cases(dat), ]

  # for genes associated with multiple sites, use the log2ratio with the largest magnitude
  max_abs <- function(x) {
    x[which.max(abs(x))]
  }

  dat <- dat %>%
    group_by(symbol) %>%
    summarize(across(everything(), max_abs))

  dat
}
```

## Nutrient-fed

```{r load_nutri_data, message=FALSE}
nutri <- load_phospho_data("../data/phospho-peptides-nutri.tsv")

dim(nutri)
```

```{r preview_nutri}
# preview data
head(nutri)
```

```{r nutri_fgsea, message = FALSE, warning = FALSE}
mtorc1_nutri_genes <- nutri %>%
  select(symbol, rap_vs_control_log2ratio) %>%
  deframe() %>%
  sort(decreasing = TRUE)

mtorc2_nutri_genes <- nutri %>%
  select(symbol, ric_vs_control_log2ratio) %>%
  deframe() %>%
  sort(decreasing = TRUE)

mtorc3_nutri_genes <- nutri %>%
  select(symbol, m7_vs_control_log2ratio) %>%
  deframe() %>%
  sort(decreasing = TRUE)

set.seed(1)
mtorc1_nutri_res <- GSEA(mtorc1_nutri_genes, TERM2GENE = gene_sets, seed = TRUE,
                         pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)

set.seed(1)
mtorc2_nutri_res <- GSEA(mtorc2_nutri_genes, TERM2GENE = gene_sets, seed = TRUE,
                         pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)

set.seed(1)
mtorc3_nutri_res <- GSEA(mtorc3_nutri_genes, TERM2GENE = gene_sets, seed = TRUE,
                         pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
```

```{r mtorc1_results}
mtorc1_nutri_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

if (nrow(mtorc1_nutri_res@result) > 0) {
  dotplot(mtorc1_nutri_res) +
    ggplot2::xlim(0, 1.1)
}
```

```{r mtorc2_nutri_results}
mtorc2_nutri_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

if (nrow(mtorc2_nutri_res@result) > 0) {
  dotplot(mtorc2_nutri_res) +
    ggplot2::xlim(0, 1.1)
}
```

```{r mtorc3_nutri_results}
mtorc3_nutri_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

if (nrow(mtorc3_nutri_res@result) > 0) {
  dotplot(mtorc3_nutri_res) +
    ggplot2::xlim(0, 1.1)
}
```

## Starved

```{r load_starved_data, message=FALSE}
starved <- load_phospho_data("../data/phospho-peptides-starved.tsv")

dim(starved)
```

```{r preview_starved}
head(starved)
```

```{r fgsea_starved, message = FALSE, warning = FALSE}
mtorc1_starved_genes <- starved %>%
  select(symbol, rap_vs_control_log2ratio) %>%
  deframe() %>%
  sort(decreasing = TRUE)

mtorc2_starved_genes <- starved %>%
  select(symbol, ric_vs_control_log2ratio) %>%
  deframe() %>%
  sort(decreasing = TRUE)

mtorc3_starved_genes <- starved %>%
  select(symbol, m7_vs_control_log2ratio) %>%
  deframe() %>%
  sort(decreasing = TRUE)

set.seed(1)
mtorc1_starved_res <- GSEA(mtorc1_starved_genes, TERM2GENE = gene_sets, seed = TRUE,
                           pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)

set.seed(1)
mtorc2_starved_res <- GSEA(mtorc2_starved_genes, TERM2GENE = gene_sets, seed = TRUE,
                           pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)

set.seed(1)
mtorc3_starved_res <- GSEA(mtorc3_starved_genes, TERM2GENE = gene_sets, seed = TRUE,
                           pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
```

```{r mtorc1_starved_results}
mtorc1_starved_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

if (nrow(mtorc1_starved_res@result) > 0) {
  dotplot(mtorc1_starved_res) +
    ggplot2::xlim(0, 1.1)
}
```

```{r mtorc2_starved_results}
mtorc2_starved_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

if (nrow(mtorc2_starved_res@result) > 0) {
  dotplot(mtorc2_starved_res) +
    ggplot2::xlim(0, 1.1)
}
```

```{r mtorc3_starved_results}
mtorc3_starved_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

if (nrow(mtorc3_starved_res@result) > 0) {
  dotplot(mtorc3_starved_res) +
    ggplot2::xlim(0, 1.1)
}
```

## Save results

```{r save_results}
if (!dir.exists(out_dir)) {
  dir.create(out_dir, mode = "0755")
}

# save individual tsv files
write_tsv(mtorc1_nutri_res@result, file.path(out_dir, "phosphoproteomics-gsea-results-nutri_mtorc1.tsv"))
write_tsv(mtorc2_nutri_res@result, file.path(out_dir, "phosphoproteomics-gsea-results-nutri_mtorc2.tsv"))
write_tsv(mtorc3_nutri_res@result, file.path(out_dir, "phosphoproteomics-gsea-results-nutri_mtorc3.tsv"))

write_tsv(mtorc1_starved_res@result, file.path(out_dir, "phosphoproteomics-gsea-results-starved_mtorc1.tsv"))
write_tsv(mtorc2_starved_res@result, file.path(out_dir, "phosphoproteomics-gsea-results-starved_mtorc2.tsv"))
write_tsv(mtorc3_starved_res@result, file.path(out_dir, "phosphoproteomics-gsea-results-starved_mtorc3.tsv"))

# save combined xls file
xlsx_outfile <- file.path(out_dir, "phosphoproteomics-gsea-results.xlsx")

write.xlsx(mtorc1_nutri_res@result, file = xlsx_outfile, sheetName = "mTORC1-nutri")

if (nrow(mtorc1_starved_res@result) > 0) {
  write.xlsx(mtorc1_starved_res@result, file = xlsx_outfile, sheetName = "mTORC1-starved", append = TRUE)
}
if (nrow(mtorc2_nutri_res@result) > 0) {
  write.xlsx(mtorc2_nutri_res@result, file = xlsx_outfile, sheetName = "mTORC2-nutri", append = TRUE)
}
if (nrow(mtorc2_starved_res@result) > 0) {
  write.xlsx(mtorc2_starved_res@result, file = xlsx_outfile, sheetName = "mTORC2-starved", append = TRUE)
}
if (nrow(mtorc3_nutri_res@result) > 0) {
  write.xlsx(mtorc3_nutri_res@result, file = xlsx_outfile, sheetName = "mTORC3-nutri", append = TRUE)
}
if (nrow(mtorc3_starved_res@result) > 0) {
  write.xlsx(mtorc3_starved_res@result, file = xlsx_outfile, sheetName = "mTORC3-starved", append = TRUE)
}
```

## Session Info

```{r session_info}
sessionInfo()
```
