---
title: "CRISPR-guided multiomics unveils direct mTOR inhibition of Hippo signaling through CDCP1: Transcriptomics GSEA Analysis"
output:
  html_document:
    df_print: kable
  pdf_document:
    df_print: kable
---

## Overview

Here, gene set enrichment analysis (GSEA) is performed for each of the three mTOR
complex knockouts, with genes weighted and ranked based on the ratio of average gene
expression for the knockouts compared to controls.

This analysis was performed separately for nutrient-fed and starved samples, and was
performed for both RNA-Seq data (here), and phosphoproteomics data (separate file.)

To perform the actual gene set enrichment analysis, the `GSEA()` function from the
[clusterProfiler](https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html) 
Bioconductor package is used.

`GSEA()` provides a helpful wrapper around [Fast Gene Set Enrichment Analysis
(fgsea)](https://bioconductor.org/packages/release/bioc/html/fgsea.html), and makes it
easy to visualize the results of enrichment analyses.

All of the code used to perform this analysis, along with the resulting tables and
figures depicting significantly enriched gene sets are shown in this [RMarkdown](https://rmarkdown.rstudio.com/) 
document.

The full source .Rmd as well as the input datasets used to perform the analysis can be
found at: [https://github.com/khughitt/nguyen22-crispr-guided-multiomics](https://github.com/khughitt/nguyen22-crispr-guided-multiomics)

## Setup

```{r message = FALSE}
library(clusterProfiler)
library(enrichplot)
library(ggnewscale)
library(msigdbr)
library(tidyverse)
library(xlsx)

# plot opts
knitr::opts_chunk$set(
  fig.width = 6,
  fig.height = 3.6,
  fig.retina = 1,
  dpi = 192
)

# output directory?
out_dir <- "output/"

FGSEA_NPERM <- 10000
MIN_PADJ <- 0.05
```

```{r load_gene_sets}
# load gene MSIGdb gene sets
gene_sets <- msigdbr(category = "H")

gene_sets <- gene_sets %>%
  select(gs_name, gene_symbol) %>%
  distinct()
```

## Nutrient-fed

First, we load the nutrient-fed RNA-Seq counts and perform a simple CPM size-factor
normalization to enable sample-wise comparisons.

Next, mean expression for each gene and for each genotype is computed.

Then, the ratio of each knockout condition to the control levels is computed. When
computing the ratios, one is added to both sides of the equation to avoid divide-by-zero
errors in cases where the control samples had no measurable expression of a gene.

```{r load_nutri_data, message=FALSE}
nutri <- read_tsv('../data/rnaseq_nutri.tsv', show_col_types = FALSE)

# perform size factor normalization
ind <- grepl("_[1-4]$", colnames(nutri))
nutri[, ind] <- sweep(nutri[, ind], 2, colSums(nutri[, ind]), '/') * 1E6

dim(nutri)

# compute condition-specific totals
nutri_rap <- nutri %>%
  select(contains("Rap")) %>%
  rowMeans()

nutri_ric <- nutri %>%
  select(contains("Ric")) %>%
  rowMeans()

nutri_m7 <- nutri %>%
  select(contains("m7")) %>%
  rowMeans()

nutri_control <- nutri %>%
  select(starts_with("v_")) %>%
  rowMeans()

# compute knockout vs. control ratios
nutri$Ratio_Rap_v_Control <- (nutri_rap + 1) / (nutri_control + 1)
nutri$Ratio_Ric_v_Control <- (nutri_ric + 1) / (nutri_control + 1)
nutri$Ratio_m7_v_Control <- (nutri_m7 + 1) / (nutri_control + 1)
```

```{r}
# preview data
nutri[1:5, 1:5]
```

Next, for each knockout and each gene symbol, log-ratios are computed. In cases where
multiple entries are present for a given symbol, due to the 1:many nature of gene
symbols & the ensembl gene identifiers originally used for mapping and quantification,
the row with the largest absolute log2 ratio is used.

The log2 ratios are sorted, and converted into 1d named vectors, as expected by
`GSEA()`.

```{r create_gene_lists_nutri}
# construct gene score vectors for GSEA analysis;
mtorc1_nutri_genes <- nutri %>%
  select(symbol, ratio = Ratio_Rap_v_Control) %>%
  mutate(log2ratio = log2(ratio)) %>%
  mutate(log2ratio_abs = abs(log2ratio)) %>%
  group_by(symbol) %>%
  arrange(-log2ratio_abs) %>%
  slice(1) %>%
  ungroup() %>%
  select(symbol, log2ratio) %>%
  arrange(-log2ratio) %>%
  deframe()

mtorc2_nutri_genes <- nutri %>%
  select(symbol, ratio = Ratio_Ric_v_Control) %>%
  mutate(log2ratio = log2(ratio)) %>%
  mutate(log2ratio_abs = abs(log2ratio)) %>%
  group_by(symbol) %>%
  arrange(-log2ratio_abs) %>%
  slice(1) %>%
  ungroup() %>%
  select(symbol, log2ratio) %>%
  arrange(-log2ratio) %>%
  deframe()

mtorc3_nutri_genes <- nutri %>%
  select(symbol, ratio = Ratio_m7_v_Control) %>%
  mutate(log2ratio = log2(ratio)) %>%
  mutate(log2ratio_abs = abs(log2ratio)) %>%
  group_by(symbol) %>%
  arrange(-log2ratio_abs) %>%
  slice(1) %>%
  ungroup() %>%
  select(symbol, log2ratio) %>%
  arrange(-log2ratio) %>%
  deframe()
```

```{r preview_top_n_gene_ratios_nutri}
head(enframe(mtorc1_nutri_genes))
head(enframe(mtorc2_nutri_genes))
head(enframe(mtorc3_nutri_genes))
```

Next, gene set enrichment is assessed using the `GSEA()` function.

```{r fgsea, message = FALSE, warning = FALSE}
mtorc1_nutri_res <- GSEA(mtorc1_nutri_genes, TERM2GENE = gene_sets, seed = 321,
                   pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
mtorc2_nutri_res <- GSEA(mtorc2_nutri_genes, TERM2GENE = gene_sets, seed = 321,
                   pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
mtorc3_nutri_res <- GSEA(mtorc3_nutri_genes, TERM2GENE = gene_sets, seed = 321,
                   pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
```

Finally, for each of the mTORC knockouts, tables and dotplots of significantly enriched
gene sets are displayed.

The dotplot visualizations show enriched gene sets along the y-axis (one per row), and
the ratio of query genes to gene set genes along the x-axis. The size of each size is
scaled in proportion to the total number of enriched genes matched, while color is used
to convey statistical significance (adjusted P-value.)

```{r mtorc1_nutri_results}
mtorc1_nutri_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

dotplot(mtorc1_nutri_res) +
  ggplot2::xlim(0, 0.8)
```

```{r mtorc2_nutri_results}
mtorc2_nutri_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

dotplot(mtorc2_nutri_res) +
  ggplot2::xlim(0, 0.8)
```

```{r mtorc3_nutri_results}
mtorc3_nutri_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

dotplot(mtorc3_nutri_res) +
  ggplot2::xlim(0, 0.8)
```

## Starved

In the following sections, a similar analysis to what is described above is performed,
but this time limited to the _starved_ samples.

```{r load_starved_data, message=FALSE}
starved <- read_tsv('data/rnaseq_starved.tsv', show_col_types = FALSE)

# perform size factor normalization
ind <- grepl("_[1-4]$", colnames(starved))
starved[, ind] <- sweep(starved[, ind], 2, colSums(starved[, ind]), '/') * 1E6

dim(starved)

# compute condition-specific totals
starved_rap <- starved %>%
  select(contains("Rap")) %>%
  rowMeans()

starved_ric <- starved %>%
  select(contains("Ric")) %>%
  rowMeans()

starved_m7 <- starved %>%
  select(contains("m7")) %>%
  rowMeans()

starved_control <- starved %>%
  select(starts_with("v_")) %>%
  rowMeans()

# compute knockout vs. control ratios
starved$Ratio_Rap_v_Control <- (starved_rap + 1) / (starved_control + 1)
starved$Ratio_Ric_v_Control <- (starved_ric + 1) / (starved_control + 1)
starved$Ratio_m7_v_Control <- (starved_m7 + 1) / (starved_control + 1)
```

```{r}
# preview data
starved[1:5, 1:5]
```

```{r create_gene_lists_starved}
# construct gene score vectors for GSEA analysis;
mtorc1_starved_genes <- starved %>%
  select(symbol, ratio = Ratio_Rap_v_Control) %>%
  mutate(log2ratio = log2(ratio)) %>%
  mutate(log2ratio_abs = abs(log2ratio)) %>%
  group_by(symbol) %>%
  arrange(-log2ratio_abs) %>%
  slice(1) %>%
  ungroup() %>%
  select(symbol, log2ratio) %>%
  arrange(-log2ratio) %>%
  deframe()

mtorc2_starved_genes <- starved %>%
  select(symbol, ratio = Ratio_Ric_v_Control) %>%
  mutate(log2ratio = log2(ratio)) %>%
  mutate(log2ratio_abs = abs(log2ratio)) %>%
  group_by(symbol) %>%
  arrange(-log2ratio_abs) %>%
  slice(1) %>%
  ungroup() %>%
  select(symbol, log2ratio) %>%
  arrange(-log2ratio) %>%
  deframe()

mtorc3_starved_genes <- starved %>%
  select(symbol, ratio = Ratio_m7_v_Control) %>%
  mutate(log2ratio = log2(ratio)) %>%
  mutate(log2ratio_abs = abs(log2ratio)) %>%
  group_by(symbol) %>%
  arrange(-log2ratio_abs) %>%
  slice(1) %>%
  ungroup() %>%
  select(symbol, log2ratio) %>%
  arrange(-log2ratio) %>%
  deframe()
```

```{r preview_top_n_gene_ratios_starved}
head(enframe(mtorc1_starved_genes))
head(enframe(mtorc2_starved_genes))
head(enframe(mtorc3_starved_genes))
```

```{r fgsea, message = FALSE, warning = FALSE}
mtorc1_starved_res <- GSEA(mtorc1_starved_genes, TERM2GENE = gene_sets, seed = 321,
                   pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
mtorc2_starved_res <- GSEA(mtorc2_starved_genes, TERM2GENE = gene_sets, seed = 321,
                   pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
mtorc3_starved_res <- GSEA(mtorc3_starved_genes, TERM2GENE = gene_sets, seed = 321,
                   pvalueCutoff = MIN_PADJ, nPermSimple = FGSEA_NPERM)
```

```{r mtorc1_starved_results}
mtorc1_starved_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

dotplot(mtorc1_starved_res) +
  ggplot2::xlim(0, 0.8)
```

```{r mtorc2_starved_results}
mtorc2_starved_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

dotplot(mtorc2_starved_res) +
  ggplot2::xlim(0, 0.8)
```

```{r mtorc3_starved_results}
mtorc3_starved_res@result %>%
  select(setSize, NES, pvalue, p.adjust)

dotplot(mtorc3_starved_res) +
  ggplot2::xlim(0, 0.8)
```

## Save results

```{r save_results}
if (!dir.exists(out_dir)) {
  dir.create(out_dir, mode = "0755")
}

# save individual tsv files
write_tsv(mtorc1_nutri_res@result, file.path(out_dir, "transcriptomics-gsea-results-nutri-mtorc1.tsv"))
write_tsv(mtorc2_nutri_res@result, file.path(out_dir, "transcriptomics-gsea-results-nutri-mtorc2.tsv"))
write_tsv(mtorc3_nutri_res@result, file.path(out_dir, "transcriptomics-gsea-results-nutri-mtorc3.tsv"))

write_tsv(mtorc1_starved_res@result, file.path(out_dir, "transcriptomics-gsea-results-starved-mtorc1.tsv"))
write_tsv(mtorc2_starved_res@result, file.path(out_dir, "transcriptomics-gsea-results-starved-mtorc2.tsv"))
write_tsv(mtorc3_starved_res@result, file.path(out_dir, "transcriptomics-gsea-results-starved-mtorc3.tsv"))

# save combined xls file
xlsx_outfile <- file.path(out_dir, "transcriptomics-gsea-results.xlsx")

write.xlsx(mtorc1_nutri_res@result, file = xlsx_outfile, sheetName = "mTORC1-nutri")
write.xlsx(mtorc1_starved_res@result, file = xlsx_outfile, sheetName = "mTORC1-starved", append = TRUE)
write.xlsx(mtorc2_nutri_res@result, file = xlsx_outfile, sheetName = "mTORC2-nutri", append = TRUE)
write.xlsx(mtorc2_starved_res@result, file = xlsx_outfile, sheetName = "mTORC2-starved", append = TRUE)
write.xlsx(mtorc3_nutri_res@result, file = xlsx_outfile, sheetName = "mTORC3-nutri", append = TRUE)
write.xlsx(mtorc3_starved_res@result, file = xlsx_outfile, sheetName = "mTORC3-starved", append = TRUE)
```

## Session Info

```{r}
sessionInfo()
```
